pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/data/entity/User.java

```diff
@@ -81,6 +81,10 @@ public User setSubject(String subject) {
 	}
 
 	public Set<Role> getRoles() {

```

Comment:
Why not set roles to an empty set when you instantiate the entity? You've created a race condition here.

Reply:
I didn't know this could create a race condition. I read up on the topic a little. I have removed this and made necessary code changes.